Code/Framework/AtomCore/AtomCore/Utils/ScopedValue.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/std/functional.h>
+
+namespace AZ
+{
+    //! Sets a variable upon construction and again when the object goes out of scope.
+    template<typename T>
+    class ScopedValue
+    {
+    private:
+        T* m_ptr;
```

Comment:
Would this be safer as a reference? If not, should you assert that the pointer is not null in the constructor?

Reply:
Added an assert.

Reply:
(I prefer pointers as the call site is more explicit in what is going on)